src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -2532,6 +2532,29 @@ static QVariant fcnConvexHull( const QVariantList &values, const QgsExpressionCo
   QVariant result = !geom.isNull() ? QVariant::fromValue( geom ) : QVariant();
   return result;
 }
+
+static QVariant fcnMinimalCircle( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  QgsGeometry fGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  QgsPointXY center;
+  double radius;
+  unsigned int segments = 36;
+  if ( values.length() == 2 )
+    segments = QgsExpressionUtils::getDoubleValue( values.at( 1 ), parent );
+  QgsGeometry geom = fGeom.minimalEnclosingCircle( center, radius, segments );
+  QVariant result = !geom.isNull() ? QVariant::fromValue( geom ) : QVariant();
+  return result;
+}
+
+static QVariant fcnOrientedBBox( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  QgsGeometry fGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  double area, angle, width, height;
+  QgsGeometry geom = fGeom.orientedMinimumBoundingBox( area, angle, width, height );
+  QVariant result = !geom.isNull() ? QVariant::fromValue( geom ) : QVariant();
```

Comment:
Is there a problem with just returning a QVariant (geom ) also for null?

Reply:
We've got lots of code which just tests variant.isValid() as a way of testing for null. Returning an empty variant is safer than an empty geometry (which will return true for isValid)

Reply:
Yes. For these expressions I used the same logic as for the convex hull's expression.